### PR TITLE
CAS2-416: Allow Licence CA's to see prison dashboard

### PR DESCRIPTION
**Pull request by Harriethw**
# Context

CAS2 currently is setup for users with the POM
role to be able to create/read applications. We
have since found that the ROLE_POM is given out to
a broad range of users who may not be POMs - The
[LICENCE_CA role](https://dsdmoj.atlassian.net/wiki/spaces/CVL/pages/3540320445/Users+and+Roles) is more in line with who our
users are, and is also used by HDC services to
perform similar actions that we require. We are
therefore going to add in the LICENCE_CA role and
remove the POM role permissions in stages.

This is the first stage - to allow them to view submitted applications for their prisons.

# Changes in this PR

Having done the work on [the API to allow read
access to LICENCE_CA]( https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/1786) this PR allows the users to see the prison dashboard tile.

A User with the LICENCE_CA role has been created for AP Tools locally, and I've setup one for the dev/test environment under our 1Password vault as 'DEV: CAS2 Licence CA user'

### After

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Are any changes required to the e2e tests?
- [ ] If you've added a new route, have you added a new
  `auditEvent`? (see `server/routes/apply.ts` for examples)
- [ ] Are there environment variables or other infrastructure configuration which needs to be included in this release?
- [ ] Are there any data migrations required. Automatic or manual?
- [ ] Does this rely on changes being deployed to the CAS API?

## Post merge checklist

Once we've merged it will be auto-deployed to the dev environment.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-community-accommodation-tier-2-ui).

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

File: server/utils/userUtils.test.ts
Comment (patrickjfl) [inline server/utils/userUtils.test.ts:33]: would it make sense to remove the feature flag before merging this branch?

**Review by patrickjfl — APPROVED** [review]
Looks OK to me. Are we going to end up removing the POM role altogether though? If so just wondering how much value there is in the incremental approach vs big bang.

File: server/utils/userUtils.test.ts
Comment (Harriethw) [reply server/utils/userUtils.test.ts:33]: ah that's a good point yeah, will take a look at your PR

**Comment by Harriethw** [comment]
@patrickjfl I think the incremental approach gives us some value now - the LCA's can then see prison dashboard ASAP. It's going to take me at least a day or two to give them write access as well. It's also more work to also remove the POM role (and that could potentially be a big change so we might want to get a bit more data on our users and be prepared to roll back when we remove that).

**Comment by patrickjfl** [comment]
> @patrickjfl I think the incremental approach gives us some value now - the LCA's can then see prison dashboard ASAP. It's going to take me at least a day or two to give them write access as well (because it needs changes to API). It's also more work to also remove the POM role (and that could potentially be a big change so we might want to get a bit more data on our users and be prepared to roll back when we remove that).

Ah ok - I was under the impression that everyone with role LCA also has role POM, in which case they'd all be able to see the prison dashboard anyway, but is that not the case?

**Comment by Harriethw** [comment]
> Ah ok - I was under the impression that everyone with role LCA also has role POM, in which case they'd all be able to see the prison dashboard anyway, but is that not the case?

I don't think there's any actual hard data for this! It's possible an LCA might not have role POM, according to an email and technically.
Having said that the API work has been a lot less than I expected to might just do all the UI in one go after all :)

**Comment by Harriethw** [comment]
> Having said that the API work has been a lot less than I expected to might just do all the UI in one go after all :)
actually the API has gotta go through review etc. so might as well merge this as it's ready

**Comment by patrickjfl** [comment]
> > Having said that the API work has been a lot less than I expected to might just do all the UI in one go after all :)
> 
> actually the API has gotta go through review etc. so might as well merge this as it's ready

👍

**Pull request merged**
